doc/source/building/compilers_and_options.rst

```diff
@@ -0,0 +1,118 @@
+Compiler selection and customizing a build
+==========================================
+
+Selecting a specific compiler
+-----------------------------
+
+Meson supports the standard environment variables ``CC``, ``CXX`` and ``FC`` to
+select specific C, C++ and/or Fortran compilers. These environment variables are
+documented in `the reference tables in the Meson docs
+<https://mesonbuild.com/Reference-tables.html#compiler-and-linker-flag-environment-variables>`__.
+
+Note that environment variables only get applied from a clean build, because
+they affect the configure stage (i.e., ``meson setup``). An incremental rebuild
+does not react to changes in environment variables - you have to run ``git
+clean -xdf`` and do a full rebuild, or run ``meson setup --reconfigure``.
+
+
+Adding a custom compiler or linker flag
+---------------------------------------
+
+Meson by design prefers builds being configured through command-line options
+passed to ``meson setup``. It provides many built-in options:
+
+- For enabling a debug build and the optimization level, see the next section
+  on "build types",
+- Enabling ``-Werror`` in a portable manner is done via ``-Dwerror=true``,
+- Enabling warning levels is done via ``-Dwarning_level=<int>``,
+- There are many other builtin options, from activating Visual Studio (``-D--vsenv``)
```

Comment:
Just --vsenv. The whole "pass a --flag option to another option" thing is part of the pip/build indirection layer. :D

Reply:
Thanks. I see that it can be `-Dvsenv=true` since Meson 1.1.0 (https://mesonbuild.com/Builtin-options.html#details-for-vsenv). That may be preferable for consistency?

Reply:
Had a look at the upstream change: https://github.com/mesonbuild/meson/pull/11620 says that `-Dvsenv=true` has more consistent behavior when reconfiguring, so I'll go with that. We already require Meson >=1.1.0, so it's fine from that perspective.

Reply:
It doesn't have more consistent behavior when reconfiguring -- meson 1.1.0 has more consistent behavior when reconfiguring, because --vsenv is now saved as a builtin option.

On 1.1.0, --vsenv and -Dvsenv=true are parsed the same, and for the same reason that `--buildtype release` and `-Dbuildtype=release` are parsed the same.

Reply:
Ah, makes sense, thanks. Using `-D` still seems to make sense for consistency, given that we use `-Dbuildtype` and other `-D` options too.